src/build-config.ts

```diff
@@ -373,6 +371,58 @@ export async function buildEntryConfig(
     configs.push(...buildConfigs)
   })
 
+  if (pkg.bin) {
+    const isSinglePathBin = typeof pkg.bin === 'string'
+    const binDistPaths = isSinglePathBin
+      ? [pkg.bin as string]
+      : Object.values(pkg.bin)
+
+    for (const binDistPath of binDistPaths) {
```

Comment:
One idea is we could have a fixed folder `src/bin` to map all the bindary files, and that folder is reserved. Then it's easier to let users remember the constraints and the path mapping here will be much easier. Then we can document it in the readme

Reply:
Sounds good!